### PR TITLE
Fixed chemprop/train/molecule_fingerprint.py to be able to take as input additional atom descripotrs for MPN version of fingerprint

### PR DESCRIPTION
**Pull request by MariBerry**
This fixes issue #233  by altering number of dimensions in the final layer (for --descriptors descriptor) or in the input atom features (for --descriptors feature)

### CODE REVIEW

**Comment by cjmcgill** [comment]
Hi @MariBerry, thank you for your engagement! Sorry that we didn't get a chance to dig into your issue. Your updating of the fingerprint code looks great. There's one recommendation I would make that I'll mark in review and after that we can push it out.

File: chemprop/train/molecule_fingerprint.py [hunk 1]
```diff
@@ -76,7 +76,12 @@ def molecule_fingerprint(args: FingerprintArgs, smiles: List[List[str]] = None)
 
     # Set fingerprint size
     if args.fingerprint_type == 'MPN':
-        total_fp_size = args.hidden_size * args.number_of_molecules
+        if args.atom_descriptors == "descriptor": # special case when we have 'descriptor' extra dimensions need to be added
+            total_fp_size = (args.hidden_size + test_data.atom_descriptors_size()) * args.number_of_molecules
+        else:
+                total_fp_size = args.hidden_size * args.number_of_molecules
+        if args.atom_descriptors == "feature":
```
Comment (cjmcgill) [inline chemprop/train/molecule_fingerprint.py:83]: There is another bit that could be added to make it compatible with extra bond features too. Could you remove your new lines 83 and 84 and then add these lines from `make_predictions/set_features` between lines 38 and 39?

```
    if args.atom_descriptors == 'feature':
        set_extra_atom_fdim(train_args.atom_features_size)

    if args.bond_features_path is not None:
        set_extra_bond_fdim(train_args.bond_features_size)
```

**Comment by MariBerry** [comment]
Hi @cjmcgill , I added  those lines

File: chemprop/train/molecule_fingerprint.py [hunk 2]
```diff
@@ -36,6 +36,12 @@ def molecule_fingerprint(args: FingerprintArgs, smiles: List[List[str]] = None)
 
     #set explicit H option and reaction option
     reset_featurization_parameters()
+    if args.atom_descriptors == 'feature':
+        set_extra_atom_fdim(train_args.atom_features_size)
+
+    if args.bond_features_path is not None:
+        set_extra_bond_fdim(train_args.bond_features_size)
```
Comment (cjmcgill) [inline chemprop/train/molecule_fingerprint.py:43]: @MariBerry sorry, I set you up for a problem here. You also need to import `set_extra_bond_fdim`

**Review by cjmcgill — APPROVED** [review]
Passes testing and looks good. Thanks for your engagement!

**Pull request merged**